utils/hare-fetch-fids

```diff
@@ -0,0 +1,202 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+
+# :help: Fetch the fids for services configured on a node
+
+import sys
+import re
+from typing import List, NamedTuple
+import json
+import argparse
+import subprocess
+import simplejson
+from hax.types import ObjT
+
+
+class Node:
+    def __init__(self, name: str):
+        self.name = name
+        self.svcs = []
+
+    def add_service(self, svc: str, fid: str):
+        self.svcs.append(Process(svc, fid))
+
+    def get_service(self, service: str):
+        for svc in self.svcs:
+            if svc.name == service:
+                return svc.fid
+        print('Service not found', file=sys.stderr)
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}(name={self.name}, svcs={self.svcs})'
+
+    def for_json(self):
+        return self.__dict__
+
+
+Process = NamedTuple('Process', [('name', str), ('fid', str)])
+
+
+def to_fid(container: int, key: int) -> str:
+    return f'{container:#x}:{key:#x}'
+
+
+def node_names(data: List) -> List[Node]:
+    # Get entries from file
+    # ​"key": "m0conf/nodes/<node_fid>",
+    # "value": "{​​​​​\"name\": \"<node name>\", \"state\": \"<HA state>\"}​​​​​"
+    regex = re.compile(
+        f'm0conf/nodes/[a-zA-Z0-9:]+$')
+    nodes = []
+    for key in data:
+        match_result = re.match(regex, key['key'])
+        if not match_result:
+            continue
+        if len(key['key'].split('/')) == 3:
+            nodes.append(json.loads(key['value'])['name'])
+    return nodes
+
+
+def get_keys_for_processes(data: List) -> List[str]:
+    # Get 'm0conf/nodes/<node_id>/processes/<process_fidk>/services/<svc_type>' entries
+    # from the file.  See 'Consul KV Schema' in [4/KV](rfc/4/README.md).
+    regex = re.compile(
+        f'^m0conf\\/.*\\/processes\\/.*\\/services\\/(confd|ios|m0_client_s3)+$')
+
+    keys = []
+    for key in data:
+        match_result = re.match(regex, key['key'])
+        if not match_result:
+            continue
+        keys.append(match_result.group(0))
+    if not keys:
+        raise RuntimeError(
+            f'No ioservice, confd, or s3servers found in the cluster')
+    return keys
+
+
+def add_service_to_node(nodes: List[Node], node_name: str, svc_name: str, fid: str):
+    for node in nodes:
+        if node.name == node_name:
+            node.add_service(svc_name, fid)
+
+
+def get_service_for_node(nodes: List[Node], node_name: str = None, service: str = None) -> str:
+    # if node_name is not specified, then local node is used.
+    if node_name == None:
+        process = subprocess.run("node-name",
```

Comment:
It would be good to add this as an utility function in utils/utils.py.

Reply:
Oh sure, makes sense

Reply:
done

Reply:
We also need unit test, for entire hctl interface for that matter. Let's move this task to that patch.